bundles/org.openhab.binding.onewire/src/test/java/org/openhab/binding/onewire/device/DS2438Test.java

```diff
@@ -53,6 +57,13 @@ public void setupMocks() {
     public void temperatureChannel() {
         instantiateDevice();
 
+        final AbstractOwDevice testDevice = this.testDevice;
```

Comment:
It would be really nice to have a different solution for this repeated pattern. How does `Objects.requireNonNull()` work with our checks?

Reply:
True. I did these changes to get rid of the "potential null-pointer access" warnings. Using `Objects.requireNonNull()` does not help much, I can only replace the if-clause, the assignment to the local variables is still needed. Unfortunately I can't make the fields final because they change from test to test. Another workaround would be a NonNullByDefault-DTO which contains the `inOrder` and `testDevice` objects and is returned by `instantiateDevice()`

Reply:
I would remove `testDevice` field and make `instantiateDevice()` to return NonNull as it already has Assert.fail, so in case of device creation error calling test will fail. 
This way you could remove all `testDevice` null checks.

The same for inOrder - you don't need this instance field.